app/upgrades/v7/upgrades.go

```diff
@@ -0,0 +1,322 @@
+	cellarfeeskeeper "github.com/peggyjv/sommelier/v7/x/cellarfees/keeper"
+	cellarfeestypes "github.com/peggyjv/sommelier/v7/x/cellarfees/types"
+	pubsubkeeper "github.com/peggyjv/sommelier/v7/x/pubsub/keeper"
+	pubsubtypes "github.com/peggyjv/sommelier/v7/x/pubsub/types"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	auctionKeeper auctionkeeper.Keeper,
+	cellarfeesKeeper cellarfeeskeeper.Keeper,
+	pubsubKeeper pubsubkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("v7 upgrade: entering handler")
+
+		// We must manually run InitGenesis for incentives, pubsub, and auctions so we can adjust their values
+		// during the upgrade process. RunMigrations will migrate to the new cork version. Setting the consensus
+		// version to 1 prevents RunMigrations from running InitGenesis itself.
+		fromVM[auctiontypes.ModuleName] = mm.Modules[auctiontypes.ModuleName].ConsensusVersion()
+		fromVM[pubsubtypes.ModuleName] = mm.Modules[pubsubtypes.ModuleName].ConsensusVersion()
+
+		// Params values were introduced in this upgrade but no migration was necessary, so we initialize the
+		// new values to their defaults
+		ctx.Logger().Info("v7 upgrading: setting cellarfees default params")
+		cellarfeesKeeper.SetParams(ctx, cellarfeestypes.DefaultParams())
+
+		ctx.Logger().Info("v7 upgrade: initializing auction genesis state")
+		auctionInitGenesis(ctx, auctionKeeper)
+
+		ctx.Logger().Info("v7 upgrade: initializing pubsub genesis state")
+		pubsubInitGenesis(ctx, pubsubKeeper)
+
+		ctx.Logger().Info("v7 upgrade: running migrations and exiting handler")
+		return mm.RunMigrations(ctx, configurator, fromVM)
+	}
+}
+
+// Initialize the auction module with prices for some stablecoins and SOMM.
+func auctionInitGenesis(ctx sdk.Context, auctionKeeper auctionkeeper.Keeper) {
+	genesisState := auctiontypes.DefaultGenesisState()
+
+	usomm52WeekLow := sdk.MustNewDecFromStr("0.079151")
+	oneDollar := sdk.MustNewDecFromStr("1.0")
+	// TODO(bolten): update LastUpdatedBlock to the upgrade height when finalized
+	usommPrice := auctiontypes.TokenPrice{
+		Denom:            "usomm",
+		UsdPrice:         usomm52WeekLow,
+		Exponent:         6,
+		LastUpdatedBlock: 1,
+	}
+
+	// setting stables to 1 dollar
+	usdcPrice := auctiontypes.TokenPrice{
+		Denom:            "gravity0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+		UsdPrice:         oneDollar,
+		Exponent:         6,
+		LastUpdatedBlock: 1,
+	}
+
+	usdtPrice := auctiontypes.TokenPrice{
+		Denom:            "gravity0xdAC17F958D2ee523a2206206994597C13D831ec7",
+		UsdPrice:         oneDollar,
+		Exponent:         6,
+		LastUpdatedBlock: 1,
+	}
+
+	daiPrice := auctiontypes.TokenPrice{
+		Denom:            "gravity0x6B175474E89094C44Da98b954EedeAC495271d0F",
+		UsdPrice:         oneDollar,
+		Exponent:         18,
+		LastUpdatedBlock: 1,
+	}
+
```

Comment:
We should put a BTC and ETH price in the upgrade handler

Reply:
Will do.